### PR TITLE
 General authenticate - challenge response

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Add support for GENRAL AUTHENTICATE challenge and response

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,38 +1,63 @@
...
 
 log-all = []
 log-none = []
 log-info = []
 log-debug = []
 log-warn = []
 log-error = []
+
+[patch.crates-io]
+trussed = { git = "https://github.com/sosthene-nitrokey/trussed", rev = "cbf8f3cc759fa79275fe06d3ce4661d6a4f306aa"}
```
Comment (robin-nitrokey) [inline Cargo.toml:63]: Can we use https://github.com/Nitrokey/trussed/tree/v0.1.0-nitrokey-3 instead?

**Review by robin-nitrokey — APPROVED** [review]
LGTM!

**Comment by robin-nitrokey** [comment]
Looks like `pivy-tool` is missing in the CI.

**Comment by sosthene-nitrokey** [comment]
Indeed. It's not in the debian repos either. Looks like I'll have to manually compile it 😅

**Comment by sosthene-nitrokey** [comment]
For some reason I could not compile pivy on my own laptop because it tried to load some OpenBSD compatibility shim which doesn't work, but it works in the container image so well...

**Pull request merged**